### PR TITLE
add support for fine-grained memory setting

### PR DESCRIPTION
**Pull request by goblin**
to fix https://github.com/P-H-C/phc-winner-argon2/issues/211

This is the simplest patch I could think of. Let me know whether/how I should add tests for it (i.e. should I change the way all existing tests are using m_cost, or add some new cases, or something else).

The mnemonic I have for `-k` is "KiBs of memory" - suggestions for a different letter are welcome.

### CODE REVIEW

File: src/run.c [hunk 1]
```diff
@@ -37,8 +37,8 @@
 #define UNUSED_PARAMETER(x) (void)(x)
 
 static void usage(const char *cmd) {
-    printf("Usage:  %s [-h] salt [-i|-d|-id] [-t iterations] [-m memory] "
-           "[-p parallelism] [-l hash length] [-e|-r] [-v (10|13)]\n",
+    printf("Usage:  %s [-h] salt [-i|-d|-id] [-t iterations] [-m log2(memory in KiB)] "
+           "[-k memory in KiB] [-p parallelism] [-l hash length] [-e|-r] [-v (10|13)]\n",
```
Comment (josephlr) [inline src/run.c]: Maybe do something like `[-m log2(memory in KiB) | -k memory in KiB]` to indicate that both flags should not be used at the same time. Also, should there be checking that both `-m` and `-k` are not used?

**Comment by goblin** [comment]
@josephlr I've done both, good points.

**Comment by goblin** [comment]
hang on, it doesn't seem to work ;-)

**Comment by goblin** [comment]
ok, seems to work fine, I forgot you need to specify salt before other arguments :-)

**Comment by khovratovich** [comment]
Thanks!

**Pull request merged**